lib/jira/resource/user.rb

```diff
@@ -13,10 +14,17 @@ def self.singular_path(client, key, prefix = '/')
       # This method is a bit of a hack. There is no way to get a list of all users in Jira, so we concatenate all of the project IDs
       # and use the assignable users endpoint to get a list of users that can be assigned to all projects.
       def self.all(client)
-        project_list = client.Project.all.map(&:key).join(',')
-        response = client.get("#{client.options[:rest_base_path]}/user/assignable/multiProjectSearch?projectKeys=#{project_list}&maxResults=1000")
-        users = parse_json(response.body)
-        users.map do |user|
+        all_users           = []
+        grouped_project_ids = client.Project.all.map(&:key).uniq.each_slice(SMALLER_SAMPLE_SIZE).to_a
+
+        grouped_project_ids.each do |project_ids|
+          project_list = project_ids.join(",")
+          response     = client.get("#{client.options[:rest_base_path]}/user/assignable/multiProjectSearch?projectKeys=#{project_list}&maxResults=1000")
```

Comment:
@arincon9 following up from @amandamholl's previous comment, what do you think about the `SMALLER_SAMPLE_SIZE` for `maxResults` here? Have we tested that we can get more than 100 users? Or if so, what about >100 users per project? It would be good to understand the limits here, because it feels like we're kind of kicking the can down the road and may run into another limit.

That said, what we have here gets us over the current hurdle, and other than the above questions it looks good to me.